sympy/solvers/solvers.py

```diff
@@ -350,7 +350,11 @@ def check_assumptions(expr, against=None, **assumptions):
         if test is expected:
             continue
         elif test is not None:
-            return False
+            if get is True:
+                return False, key
```

Comment:
In the case you mentioned the result will be only `False`, since to get what assumption is wrong `get` must be set to `True` , by default which is `False`.
 
Also, could you please guide me writing the test cases.